tmt/steps/discover/__init__.py

```diff
@@ -502,6 +535,14 @@ def go(self, force: bool = False) -> None:
                         if test.name == result.name and test.serial_number == result.serial_number:
                             self._failed_tests[test_phase].append(test)
 
+        # Assert that all required tests were discovered
+        for phase_name, required_test_names in self._required_test_names.items():
+            for required_test in required_test_names:
+                if not any(required_test == test.name for test in self._tests.get(phase_name, [])):
+                    raise tmt.utils.DiscoverError(
+                        f"Required test '{required_test}' not discovered in phase '{phase_name}'."
+                    )
```

Comment:
Would it be interesting to report tests that were discovered, but not required to be discovered? @psss any comments?